app/src/main/kotlin/de/rochefort/childmonitor/VolumeView.kt

```diff
@@ -0,0 +1,92 @@
+ * (at your option) any later version.
+ *
+ * Child Monitor is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Child Monitor. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.rochefort.childmonitor
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.View
+
+class VolumeView : View {
+    private val paint: Paint
+    private var volumeHistory: VolumeHistory? = null
+
+    constructor(context: Context?) : super(context) {
+        paint = initPaint()
+    }
+
+    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
+        paint = initPaint()
+    }
+
+    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
+        paint = initPaint()
+    }
+
+    private fun initPaint(): Paint {
+        val paint = Paint()
+        paint.color = Color.rgb(255, 127, 0)
+        return paint
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        val volumeHistory = this.volumeHistory ?: return
+        val height = height
+        val width = width
+        val size = volumeHistory.size() // Size is at most width
+        val volumeNorm = volumeHistory.volumeNorm
+        val relativeBrightness: Double = if (size > 0) {
+            val normalizedVolume = volumeHistory[size - 1]
+            Math.max(0.3, normalizedVolume)
+        } else {
+            0.3
+        }
+        val blue: Int
+        val rest: Int
+        if (relativeBrightness > 0.5) {
+            blue = 255
+            rest = (2 * 255 * (relativeBrightness - 0.5)).toInt()
+        } else {
+            blue = (255 * (relativeBrightness - 0.2) / 0.3).toInt()
+            rest = 0
+        }
+        val rgb = Color.rgb(rest, rest, blue)
+        canvas.drawColor(rgb)
+        if (size == 0) {
+            return
+        }
+        val margins = height * 0.1
+        val graphHeight = height - 2.0 * margins
+        val leftMost = Math.max(0, volumeHistory.size() - width)
+        val graphScale = graphHeight * volumeNorm
+        var xPrev = 0
+        var yPrev = (margins + graphHeight - volumeHistory[leftMost] * graphScale).toInt()
+        val length = Math.min(size, width)
+        for (xNext in 1 until length - 1) {
+            val yNext = (margins + graphHeight - volumeHistory[leftMost + xNext] * graphScale).toInt()
+            canvas.drawLine(xPrev.toFloat(), yPrev.toFloat(), xNext.toFloat(), yNext.toFloat(), paint)
+            xPrev = xNext
+            yPrev = yNext
+        }
+    }
+
+    fun setVolumeHistory(volumeHistory: VolumeHistory?) {
```

Comment:
explicit getter/setter functions are unidiomatic in kotlin

Reply:
Made it public.

Reply:
The public is redundant, members are public if they are not declared private or internal.
 (as pointed out in a different comment already